Grammar for Canvas API

symeonp opened the pull request:
Based on Ivan's comments.

ifratric commented on canvas/canvas.txt:
```diff
@@ -0,0 +1,297 @@
+
+<root root=true> = <lines count=50>
+
+!lineguard try { <line>; } catch(e) { console.log(e.message) }
```
Remove the semicolon here. Since lines already include semicolons, this results in every line ending with ;;

symeonp replied:
Fixed that, now should be good!

ifratric commented on canvas/canvas.txt:
```diff
@@ -0,0 +1,297 @@
+<new imgData> = new ImageData(<canvasint>, <canvasint>);
+
+# The button
+<new button> = document.getElementById('button');
```
As far as I can see there are no elements with id='button' in the template

symeonp replied:
You are right, I just added one as well.

ifratric commented on canvas/canvas.txt:
```diff
@@ -0,0 +1,297 @@
+<path2daddpath>.addPath(<path2daddpath>,<m>);
+
+# HTMLCanvasElement
+<new canvas> = document.getElementById('canvas');
```
Same here for canvas. There's going to be a canvas on the page but it's id is not set.

symeonp replied:
Updated this one here: https://github.com/symeonp/domato/commit/c34430de8f1b654a527021acb48f40ebe8bd45b7

symeonp commented:
Hey thanks, so I've updated those issues here: https://github.com/symeonp/domato/commit/2352c18f756d877f58c0311a0d042571bf13d740

My question, now am getting somehow that colour not found, this doesn't make sense honestly.
I've also commented out google's license cause it would as html code and added one-two new canvas functions and included cssproperties.txt. Is there anything else you can think of that needs to get improved? Am not closing the PR for the time being so I won't have to flood your project with PRs.
Thanks so much!

ifratric commented:
Thanks for working on this! 

I see where the problem is - you put !include ... after !begin lines, which means that the contents of the included lines will be treated like programming language lines instead of context free grammar rules (which they are).

symeonp commented:
My pleasure!
I did however experiment a bit at some phase and even adding them at the top of the grammar am still getting the above exception. Not quire sure what I'm missing here, thanks!

ifratric commented:
The included files also need to be copied to the same directory as the current grammar (canvas.txt), that's the only other thing I can think of.

symeonp commented:
Okay, I figured it out, somehow I had ```"<color">``` instead of ```"<color>"```, had to debug it a bit! Plus the raise clearly mentions it which I missed it heh!
That's why I was wondering how come did it work before, cool thanks!

ifratric commented:
Hey, sorry about the radio silence on my part, I was kept pretty busy by other things. I had a chance to play with it again today and I think it looks good! Merging :-)

symeonp commented:
Hey Ivan, thank you so much, that's great!
I will try to implement more functions in the near future as well.
Cheers

ifratric commented:
Thank you for working on it!
